### PR TITLE
plan, store: add unqiue index check in mock-tikv

### PR DESCRIPTION
**Pull request by hanfei1991**
If the index is unique and range is point, we can use `Get`

### CODE REVIEW

File: plan/plan_to_pb.go [hunk 1]
```diff
@@ -105,6 +105,11 @@ func (p *PhysicalIndexScan) ToPB(ctx context.Context) (*tipb.Executor, error) {
 		Columns: distsql.ColumnsToProto(columns, p.Table.PKIsHandle),
 		Desc:    p.Desc,
 	}
+	unique := p.Index.Unique
+	if len(p.Ranges) > 0 && len(p.Index.Columns) != len(p.Ranges[0].LowVal) {
```
Comment (shenli) [inline plan/plan_to_pb.go]: Add comment for this.

File: store/tikv/mock-tikv/executor.go [hunk 2]
```diff
@@ -176,24 +176,84 @@ func (e *indexScanExec) SetSrcExec(exec executor) {
...
+			handleBytes, err := codec.EncodeValue(b, handleDatum)
+			if err != nil {
+				return nil, errors.Trace(err)
+			}
+			values = append(values, handleBytes)
+		}
+	}
+
+	return values, nil
+
```
Comment (zimulala) [inline store/tikv/mock-tikv/executor.go]: Remove this blank.

File: store/tikv/mock-tikv/executor.go [hunk 3]
```diff
@@ -176,24 +176,84 @@ func (e *indexScanExec) SetSrcExec(exec executor) {
...
+	if len(b) > 0 {
+		if e.pkStatus != pkColNotExists {
+			values = append(values, b)
+		}
+	} else {
+		handle, err := decodeHandle(pair.Value)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		if e.pkStatus != pkColNotExists {
```
Comment (zimulala) [inline store/tikv/mock-tikv/executor.go]: Can we put this condition on line 233 directly?

File: plan/plan_to_pb.go [hunk 4]
```diff
@@ -86,6 +87,20 @@ func (p *PhysicalTableScan) ToPB(ctx context.Context) (*tipb.Executor, error) {
 	return &tipb.Executor{Tp: tipb.ExecType_TypeTableScan, TblScan: tsExec}, errors.Trace(err)
 }
 
+// checkUnique checks whether we can pass unique info to TiKV.
+func checkUnique(idx *model.IndexInfo, ranges []*ranger.IndexRange) bool {
```
Comment (winoros) [inline plan/plan_to_pb.go]: `checkUnique` -> `checkFullLen`?

**Comment by hanfei1991** [comment]
@winoros @zimulala PTAL

File: store/tikv/mock-tikv/executor.go [hunk 5]
```diff
@@ -176,24 +176,83 @@ func (e *indexScanExec) SetSrcExec(exec executor) {
...
+	values, b, err := tablecodec.CutIndexKeyNew(pair.Key, e.colsLen)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	if len(b) > 0 {
+		if e.pkStatus != pkColNotExists {
+			values = append(values, b)
+		}
+	} else {
+		if e.pkStatus != pkColNotExists {
```
Comment (zimulala) [inline store/tikv/mock-tikv/executor.go]: Why not use `else if` directly?

**Comment by zimulala** [comment]
LGTM

**Comment by shenli** [comment]
LGTM

**Pull request merged**